src/plot_cartogram.R

```diff
@@ -282,3 +282,253 @@ rm_facet_clip <- function(svg_in, file_out, width){
+#' @param date_start first day of focal month
+#' @param width Desired width of output plot
+#' @param height Desired height of output plot
+#' @param color_bknd Plot background color
+cartogram_ig <- function(file_svg, plot_left, plot_right, date_start, width, height, color_bknd){
+  plot_month <- lubridate::month(date_start, label = TRUE, abbr = FALSE)
+  plot_year <- lubridate::year(date_start)
+  
+  # import fonts
+  font_legend <- 'Noto Sans Mono'
+  font_add_google(font_legend)
+  showtext_opts(dpi = 300, regular.wt = 200, bold.wt = 700)
+  showtext_auto(enable = TRUE)
+  
+  text_color <- "#444444"
+    
+  # usgs logo
+  usgs_logo <- magick::image_read('in/usgs_logo.png') %>%
+    magick::image_colorize(100, text_color)
+  
+  # streamflow title
+  title_flow <- magick::image_read('in/streamflow.png')
+  
+  plot_margin <- 0.025
+  
+  # background
+  canvas <- grid::rectGrob(
+    x = 0, y = 0, 
+    width = 16, height = 9,
+    gp = grid::gpar(fill = color_bknd, alpha = 1, col = color_bknd)
+  )
+  
+  # Restyle legend
+  plot_left <- plot_left +
+    guides(fill = guide_colorsteps(
+      title = "",
+      nrow = 1,
+      direction = 'horizontal',
+      label.position = "bottom",
+      barwidth = 12,
+      barheight = 0.6,
+      background = element_rect(fill = NA),
+      show.limits = TRUE,
+      even.steps = FALSE
+    )) +
+    theme(legend.background = element_rect(fill = NA),
+          text = element_text(family = font_legend, color = text_color, size = 6.5))
+  
+  # Extract from plot
+  plot_legend <- get_legend(plot_left)
+  
+  # compose final plot
+  ggdraw(ylim = c(0,1), 
+         xlim = c(0,1)) +
+    # a white background
+    draw_grob(canvas,
+              x = 0, y = 1,
+              height = 0.37, width = 0.37,
+              hjust = 0, vjust = 1) +
+    # state tiles
+    draw_plot(plot_right+theme(text = element_text(family = font_legend, color = text_color),
+                               strip.text = element_text(size = 6, vjust = -3)),
+              x = 1.09,
+              y = -0.05,
+              height = 1.27,
+              width = 1.17,
+              # height = 1- plot_margin*0.01, 
+              # width = 2-(0.1+plot_margin*0.01),
+              hjust = 1,
+              vjust = 0) + 
+    # draw title
+    draw_label(sprintf('%s %s', plot_month, plot_year),
+               x = plot_margin*2, y = 1-plot_margin*1.2,
+               size = 21,
+               hjust = 0,
+               vjust = 1,
+               fontfamily = font_legend,
+               color = text_color,
+               lineheight = 1)  +
+    # stylized streamflow title
+    draw_image(title_flow,
+               x = plot_margin*2,
+               y = 1-(3*plot_margin),
+               height = 0.1,
+               width = 0.55,
+               hjust = 0,
+               vjust = 1) +
+    # add legend.
+    draw_plot(plot_legend,
+              x = plot_margin*20,
```

Comment:
Instead of using the `plot_margin` as a multiple, use relative widths to define placement. Use the `plot_margin` to define consistent spacing around the border and for adding padding around the different plot elements.

In this case, the x position ranges from 0 to 1 (left to right). That's defined above in `ggdraw`. In this case your x is 0.5 (`20*0.025`). That's setting where the upper right corner is placed because of the `hjust` and `vjust` settings here. And then you have a negative width.

That seems sort of off from what we actually see in the output though, because I would expect that the legend corner is then in the middle of the plot. There seems to be some unexpected behavior with `plot_legend` and `draw_plot`.